File: new_application_new/fillDataBase/NewsServerFill/src/main/java/models/News.java
package models;

import com.fasterxml.jackson.annotation.JsonIgnore;
import com.fasterxml.jackson.annotation.JsonProperty;


public class News {
    private  String title;
    private String author;
    private String publishedAt;
    private String url;
    private String description;
    private String content;
    private String urlToImage;
    private Source source;
    @JsonIgnore
    private boolean liked;




    public String getTitle() {
        return title;
    }

    @JsonProperty("title")
    public void setTitle(String title) {
        this.title = title;
    }

    public String getAuthor() {
        return author;
    }

    @JsonProperty("author")
    public void setAuthor(String author) {
        this.author = author;
    }

    public String getPublishedAt() {
        return publishedAt;
    }

    @JsonProperty("publishedAt")
    public void setPublishedAt(String publishedAt) {
        this.publishedAt = publishedAt;
    }

    public String getUrl() {
        return url;
    }

    @JsonProperty("url")
    public void setUrl(String url) {
        this.url = url;
    }

    public String getDescription() {
        return description;
    }

    @JsonProperty("description")
    public void setDescription(String description) {
        this.description = description;
    }

    public String getContent() {
        return content;
    }

    @JsonProperty("content")
    public void setContent(String content) {
        this.content = content;
    }

    public String getUrlToImage() {
        return urlToImage;
    }

    @JsonProperty("urlToImage")
    public void setUrlToImage(String urlToImage) {
        this.urlToImage = urlToImage;
    }


    @JsonProperty("source")
    public Source getSource() {
        return source;
    }

    public void setSource(Source source) {
        this.source = source;
    }

    public boolean isLiked() {
        return liked;
    }

    public void setLiked(boolean liked) {
        this.liked = liked;
    }
}

File: new_application_new/server/NewsServer/src/models/news.ts
export class News {
    id: number;
    title: string;
    author: string;
    publishedAt: string;
    url: string;
    description: string;
    content: string;
    urlToImage: string;
    liked: boolean;
  
    constructor(id: number, title: string,author: string, publishedAt: string,
      url: string, description: string, content: string, urlToImage: string,
      liked: boolean) {
      this.id = id;
      this.title = title;
      this.author = author;
      this.publishedAt = publishedAt;
      this.url = url;
      this.description = description;
      this.content = content;
      this.urlToImage = urlToImage;
      this.liked = liked;
    }
  }
  